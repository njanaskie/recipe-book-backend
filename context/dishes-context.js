import React from 'react';

const DishesContext = React.createContext()

export { DishesContext as default }