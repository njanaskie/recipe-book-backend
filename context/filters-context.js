import React, { useContext, useReducer } from 'react';
import filtersReducer, { filtersReducerDefaultState } from '../src/reducers/filters'

const FiltersContext = React.createContext()

export const useFiltersContext = () => useContext(FiltersContext)

const FiltersProvider = ({ children }) => {
    const [filters, filtersDispatch] = useReducer(filtersReducer, filtersReducerDefaultState)

    return (
        <FiltersContext.Provider value={{ filters, filtersDispatch }}>
            {children}
        </FiltersContext.Provider>
    )
}

export { FiltersProvider, FiltersContext as default }