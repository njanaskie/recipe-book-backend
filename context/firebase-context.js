import React, { useEffect, useState, useContext, useReducer } from 'react';
import database, { firebase, googleAuthProvider } from '../src/firebase/firebase';
import { history } from '../src/routers/AppRouter';
import authReducer from '../src/reducers/auth'
import LoadingPage from '../src/components/LoadingPage'

const FirebaseContext = React.createContext()

export const useFirebaseContext = () => useContext(FirebaseContext)

const FirebaseProvider = ({ children }) => {
    const [auth, authDispatch] = useReducer(authReducer, [])
    // const authDispatch = useDispatch()
    const [user, setUser] = useState('')
    const [loading, setLoading] = useState(true)
    const [isAdmin, setIsAdmin] = useState()
    const [isGuest, setIsGuest] = useState()
    const email = process.env.GUEST_EMAIL
    const password = process.env.GUEST_PASSWORD

    const login = () => {
        firebase.auth().signInWithPopup(googleAuthProvider).then(() => {
            authDispatch({ type: 'LOGIN' })
        })
    }

    const loginAsGuest = () => {
        firebase.auth().signInWithEmailAndPassword(email, password).then(() => {
            authDispatch({ type: 'LOGIN_AS_GUEST' })
        })
    }

    const logout = () => {
        firebase.auth().signOut().then(() => {
            authDispatch({ type: 'LOGOUT' })
        });
    }

    useEffect(() => {
        const unsubscribe = firebase.auth().onAuthStateChanged((user) => {
            setUser(user)
            setLoading(false)
            setIsAdmin()
            setIsGuest()
        })

        return () => unsubscribe()
    }, [])

    useEffect(() => {
        if (user) {
            database.collection('users').doc(user.uid).get()
            .then((doc) => {
                const userData = doc.data()
                if (userData) {
                    if (userData.isAdmin === true) {
                        const isAdmin = userData.isAdmin
                        setIsAdmin(isAdmin)
                    } else if (userData.isGuest === true) {
                        const isGuest = userData.isGuest
                        setIsGuest(isGuest)
                    }
                }
            })
        }
    }, [user])

    if (loading) {
        return <LoadingPage />
    }
    // if (user) {
    //     if (history.location.pathname === '/') {
    //         history.push('/dashboard')
    //     }
    // } else {
    //     history.push('/');
    // }

    return (
        <FirebaseContext.Provider value={{ user, isAdmin, isGuest, login, loginAsGuest, logout }}>
            {children}
        </FirebaseContext.Provider>
    )
}

export { FirebaseProvider, FirebaseContext as default }