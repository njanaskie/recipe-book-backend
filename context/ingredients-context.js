import React from 'react';

const IngredientsContext = React.createContext()

export { IngredientsContext as default }