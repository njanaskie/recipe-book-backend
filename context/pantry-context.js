import React, { useContext } from 'react';

const PantryContext = React.createContext()

export const usePantryContext = () => useContext(PantryContext)

export { PantryContext as default }