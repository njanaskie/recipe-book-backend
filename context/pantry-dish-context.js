import React, { useContext } from 'react';

const PantryDishContext = React.createContext()

export const usePantryDishContext = () => useContext(PantryDishContext)

export { PantryDishContext as default }