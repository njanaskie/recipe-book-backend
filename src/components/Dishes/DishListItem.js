import React from 'react'
import { Link } from 'react-router-dom';
import { Image } from "semantic-ui-react";

const DishListItem = ({ dish }) => {
    const pathname = window.location.pathname

    return (
        <Link to={`/dish/${dish.id}`}>
            <div className='dish-preview'>
                <div className='image-container'>
                    <Image size='medium' rounded src="/images/image-placeholder.png" />
                    <div className='dish-label'>
                        <span>{dish.name}</span>
                    </div>
                </div>
            </div>
        </Link>



    )
}

export default DishListItem