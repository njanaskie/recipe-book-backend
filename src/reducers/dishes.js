
// Dishes Reducer

const dishesReducerDefaultState = [];

const dishesReducer = (state = dishesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_DISH':
            return [
                ...state,
                action.dish
            ];
        case 'REMOVE_DISH':
            return state.filter(({ id }) => id !== action.id);
        case 'EDIT_DISH':
            return state.map((dish) => {
                if (dish.id === action.id) {
                    return {
                        ...dish,
                        ...action.updates
                    };
                } else {
                    return dish;
                };
            });
        case 'SET_DISHES':
            return action.dishes;
        default:
            return state;
    }
}

export default dishesReducer;