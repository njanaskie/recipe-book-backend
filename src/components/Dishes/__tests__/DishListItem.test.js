import React from 'react'
import { shallow } from 'enzyme'
import DishesContext from '../../../../context/dishes-context'
import * as AllDishesContext from '../../../../context/dishes-context'
import DishListItem from '../DishListItem'
import dishes from '../../../tests/fixtures/dishes'

test('should render DishListItem with dish',() => {
    // jest.spyOn(AllDishesContext, 'useDishesContext').mockImplementation(() => dishes)
    const wrapper = shallow(
            <DishListItem dish={dishes[2]}/>
        )
    expect(wrapper).toMatchSnapshot()
});