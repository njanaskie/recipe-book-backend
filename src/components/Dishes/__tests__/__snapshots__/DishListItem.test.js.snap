// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DishListItem with dish 1`] = `
<div>
  <Link
    to="/dish/3"
  >
    Perogies
  </Link>
</div>
`;
